### DISCUSSION TITLE
Sort by sequence of ids

### DISCUSSION THREAD

**Discussion by Zmaon** [mikro-orm/mikro-orm#5901] [General]
Hello,

I'm trying to sort a query by a sequence of ids, without using the query builder. In SQL (using MySQL) the query should look like this:

```SQL
select * from users WHERE id in (1, 2, 3) order by FIELD(id, 1, 2, 3);
```

I tried:

```Javascript
    return await this.userRepository.findAll({
      orderBy: raw('FIELD(id, 1, 2, 3)'),
    });
```

But is gives an error `Trying to order by not existing property User.sql`. Is this possible?

Thanks,

Simon

**Comment by B4nan** [comment]
`orderBy` needs to be an object or an array of objects. this should work:

```ts
return await this.userRepository.findAll({
  orderBy: { [raw('FIELD(id, 1, 2, 3)')]: 1, // the right hand side might be also `[]` or maybe `false` i think, not sure now
});
```

**Reply by Zmaon** [reply]
Thanks for your quick reply! Nice, this seems to work:
```Javascript
return await this.userRepository.findAll({
    orderBy: { [raw((alias) => `FIELD(${alias}.id, 1, 2, 3)`)]: 1 },
});
```
